src/internal/pachsql/db_test.go

```diff
@@ -0,0 +1,167 @@
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/jmoiron/sqlx"
+	"golang.org/x/net/context"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+
+	"github.com/DATA-DOG/go-sqlmock"
+)
+
+const (
+	collections = "collections"
+	columnName  = "column_name"
+	commits     = "commits"
+	commitDiffs = "commit_diffs"
+	key         = "key"
+	mockDBName  = "sqlmock"
+	pfs         = "pfs"
+	schemaName  = "schemaname"
+	tableName   = "tablename"
+)
+
+var (
+	commitsTable     = &SchemaTable{collections, commits}
+	commitDiffsTable = &SchemaTable{pfs, commitDiffs}
+)
+
+func TestListTables(t *testing.T) {
+	t.Run("ListTables() with a valid response", listTablesValidResponse)
+	t.Run("ListTables() with an unexpected response", listTablesUnexpectedResponse)
+	t.Run("ListTables() with an error from the database", listTablesInvalidResponse)
+}
+
+func listTablesValidResponse(t *testing.T) {
+	mockDB, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	mockRows := sqlmock.NewRows([]string{schemaName, tableName}).
+		AddRow(collections, commits).
+		AddRow(pfs, commitDiffs)
+	mock.ExpectQuery(listTablesQuery).WillReturnRows(mockRows)
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	schemaTables, err := ListTables(context.Background(), sqlxDB)
+	require.NoError(t, err, "test should pass")
+	require.Equal(t, 2, len(schemaTables),
+		"ListTables() should return same number of rows as input")
+}
+
+func listTablesUnexpectedResponse(t *testing.T) {
+	mockDB, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	mockRows := sqlmock.NewRows([]string{"food"}).
+		AddRow("sushi").
+		AddRow("cheeseburger")
+	mock.ExpectQuery(listTablesQuery).WillReturnRows(mockRows)
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	_, err = ListTables(context.Background(), sqlxDB)
+	require.Equal(t, true,
+		err != nil && strings.HasPrefix(err.Error(), "could not scan tables:"),
+		"expected an error from ListTables()")
+}
+
+func listTablesInvalidResponse(t *testing.T) {
+	mockDB, _, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	_, err = ListTables(context.Background(), sqlxDB)
+	require.Equal(t, true, err != nil && strings.HasPrefix(err.Error(), "could not list tables:"),
+		"expected an error from ListTables()")
+}
+
+func TestScannedRows(t *testing.T) {
+	t.Run("ScannedRows() with a valid response", scannedRowsValidResponse)
+	t.Run("ScannedRows() with a valid response", scannedRowsInvalidResponse)
+}
+
+func scannedRowsValidResponse(t *testing.T) {
+	mockDB, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	mockRows := sqlmock.NewRows([]string{schemaName, tableName}).
+		AddRow(collections, commits).
+		AddRow(pfs, commitDiffs)
+	mock.ExpectQuery("").WillReturnRows(mockRows)
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	rows, err := ScannedRows(context.Background(), sqlxDB, "")
+	require.NoError(t, err)
+	require.Equal(t, 2, len(rows))
```

Comment:
I am not sure length + spot check is a particularly useful test.  Just check that the rows are what you expect them to be: https://github.com/golang/go/wiki/TestComments#compare-full-structures

Reply:
Deep equals would be better